### PR TITLE
Update some project files to SDK-style.

### PR DESCRIPTION
**Pull request by jpobst**
Update `ICSharpCode.NRefactory.csproj` and `ICSharpCode.NRefactory.CSharp.csproj` to SDK style projects.  

The unit tests in Xamarin.Android use these assemblies along with the [Mono debugging assemblies](https://github.com/mono/debugger-libs) to ensure that changes do not break the ability to debug XA applications.

In order to target .NET 5/6 we need to build with `dotnet build`, which requires SDK style projects.

There does not seem to be CI for this repo.  I tested by building locally as well as running the XA CI against this change: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4453779.

### CODE REVIEW

**Comment by jpobst** [comment]
@KirillOsenkov You touched this repo last!  Can you review and merge this?  Or point me to someone who can?  Thanks!

File: ICSharpCode.NRefactory.CSharp/ICSharpCode.NRefactory.CSharp.csproj [hunk 1]
```diff
@@ -1,420 +1,43 @@
...
-    <AssemblyName>ICSharpCode.NRefactory.CSharp</AssemblyName>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <AllowUnsafeBlocks>False</AllowUnsafeBlocks>
-    <NoStdLib>False</NoStdLib>
-    <WarningLevel>4</WarningLevel>
-    <TreatWarningsAsErrors>False</TreatWarningsAsErrors>
-    <ProductVersion>8.0.30703</ProductVersion>
-    <SchemaVersion>2.0</SchemaVersion>
-    <SignAssembly>true</SignAssembly>
+    <TargetFramework>net40</TargetFramework>
```
Comment (KirillOsenkov) [inline ICSharpCode.NRefactory.CSharp/ICSharpCode.NRefactory.CSharp.csproj:4]: Any reason to target net40? Ideally it's net472.

**Review by KirillOsenkov — APPROVED** [review]
If it builds, it's good!

**Comment by jpobst** [comment]
Thanks!  `net40` was just existing I think, and no one wants to make **actual** changes to these files.  😉

**Pull request merged**